src/otx/algorithms/common/adapters/torch/dataloaders/samplers/otx_sampler.py

```diff
@@ -0,0 +1,98 @@
+"""OTX sampler."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import math
+
+from torch.utils.data import Dataset
+from torch.utils.data.sampler import Sampler
+
+from otx.algorithms.common.utils.logger import get_logger
+from otx.algorithms.common.utils.task_adapt import unwrap_dataset
+
+logger = get_logger()
+
+
+class OTXSampler(Sampler):  # pylint: disable=too-many-instance-attributes
+    """Sampler that easily adapts to the dataset statistics.
+
+    In the exterme small dataset, the iteration per epoch could be set to 1 and then it could make slow training
+    since DataLoader reinitialized at every epoch. So, in the small dataset case,
+    OTXSampler repeats the dataset to enlarge the iterations per epoch.
+
+    In the large dataset, the useful information is not totally linear relationship with the number of datasets.
+    It closes to the log scale relationship, rather.
+
+    So, this sampler samples or repeats the datasets acoording to the statistics of dataset.
+
+    Args:
+        dataset (Dataset): A built-up dataset
+        samples_per_gpu (int): batch size of Sampling
+        use_adaptive_repeats (bool): Flag about using adaptive repeats
+        num_replicas (int, optional): Number of processes participating in
+            distributed training. By default, :attr:`world_size` is retrieved from the
+            current distributed group.
+        rank (int, optional): Rank of the current process within :attr:`num_replicas`.
+            By default, :attr:`rank` is retrieved from the current distributed
+            group.
+        shuffle (bool, optional): Flag about shuffling
+        seed (int, optional): controls the randomness
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        samples_per_gpu: int,
+        use_adaptive_repeats: bool,
+        num_replicas: int = 1,
+        rank: int = 0,
+        shuffle: bool = True,
+        coef: float = -0.7,
+        min_repeat: float = 1.0,
+    ):
+
+        self.dataset, _ = unwrap_dataset(dataset)
+        self.samples_per_gpu = samples_per_gpu
+        self.num_replicas = num_replicas
+        self.rank = rank
+        self.shuffle = shuffle
+        self.repeat = self._get_proper_repeats(use_adaptive_repeats, coef, min_repeat)
+
+        self.num_samples = int(math.ceil(len(self.dataset) * self.repeat / self.num_replicas))
```

Comment:
```suggestion
        self.num_samples = math.ceil(len(self.dataset) * self.repeat / self.num_replicas)
```
It's a minor one, math.ceil returns int since python3

Reply:
Thanks for information. I just set the int casting since `self.repeat` will be float to handle the large dataset. 
For the large dataset, we don't need to use all of dataset, so will sampling by controlling the repeat (i.e. self.repeat=0.7)

Reply:
Thanks for information. Nevertheless, `self.num_samples` will be integer right? then it seems that result whether there is int casting or not is same.

Reply:
Ah, you mean that `int(9) * float(0.7) / int(1)` will be integer ?

Reply:
I mean `math.ceil(int(9) * float(0.7) / int(1))` is integer as `int(math.ceil(int(9) * float(0.7) / int(1)))` is.

Reply:
!! I mis-recognized the location of the `)`, thanks :)

Reply:
done https://github.com/openvinotoolkit/training_extensions/pull/2428/commits/a496a82861ea094fca937a3c06caff9aac85f3fd